### PR TITLE
ES-1690 adding sunbird authentication plugin

### PR DESCRIPTION
**Pull request by sacrana0**

### CODE REVIEW

File: sunbird-rc-esignet-integration-impl/README.md
Comment (ase-101) [inline sunbird-rc-esignet-integration-impl/README.md:1]: Let us rename the folder as 'sunbird-rc-plugin'

File: sunbird-rc-esignet-integration-impl/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+This library should be added as a runtime dependency to [esignet-service](https://github.com/mosip/esignet)
+
+## Configurations required to added in esignet-default.properties
+
+````
+mosip.esignet.integration.scan-base-package=io.mosip.esignet.sunbirdrc.integration
+mosip.esignet.integration.authenticator=SunbirdRCAuthenticationService
+mosip.esignet.integration.vci-plugin=SunbirdRCVCIssuancePlugin
+mosip.esignet.integration.key-binder=MockKeyBindingWrapperService
+mosip.esignet.integration.audit-plugin=LoggerAuditService
```
Comment (ase-101) [inline sunbird-rc-esignet-integration-impl/README.md:15]: We don't have to specify as these are default configurations in eSignet.

File: sunbird-rc-esignet-integration-impl/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+mosip.esignet.integration.captcha-validator=GoogleRecaptchaValidatorService
```
Comment (ase-101) [inline sunbird-rc-esignet-integration-impl/README.md:16]: We don't have to specify as these are default configurations in eSignet.

File: sunbird-rc-esignet-integration-impl/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+
+mosip.esignet.captcha-validator.url=https://www.google.com/recaptcha/api/siteverify
+mosip.esignet.captcha-validator.secret=${esignet.captcha.secret.key}
+mosip.esignet.captcha-validator.site-key=${esignet.captcha.site.key}
```
Comment (ase-101) [inline sunbird-rc-esignet-integration-impl/README.md:21]: Captcha configurations can be removed, it is also eSignet's default configuration.

File: sunbird-rc-esignet-integration-impl/README.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+mosip.esignet.vciplugin.sunbird-rc.credential-type.HealthCardCredential.cred-schema-version=1.0.0
+````
+
+
+Add "bindingtransaction" cache name in "mosip.esignet.cache.names" property.
+
+## License
+This project is licensed under the terms of [Mozilla Public License 2.0](LICENSE).
+This integration plugin is compatible with [eSignet 1.2.0](https://github.com/mosip/esignet/tree/v1.2.0) and [Sunbird-RC 1.0.0](https://github.com/Sunbird-RC/sunbird-rc-core/tree/v1.0.0)
+
```
Comment (ase-101) [inline sunbird-rc-esignet-integration-impl/README.md:52]: We should change the compatibility text.

File: sunbird-rc-esignet-integration-impl/src/main/java/io/mosip/esignet/sunbirdrc/integration/dto/ReCaptchaError.java
Comment (ase-101) [inline sunbird-rc-esignet-integration-impl/src/main/java/io/mosip/esignet/sunbirdrc/integration/dto/ReCaptchaError.java:1]: Is this file required?

File: sunbird-rc-esignet-integration-impl/src/main/java/io/mosip/esignet/sunbirdrc/integration/dto/ReCaptchaResponse.java
Comment (ase-101) [inline sunbird-rc-esignet-integration-impl/src/main/java/io/mosip/esignet/sunbirdrc/integration/dto/ReCaptchaResponse.java:1]: Is this file required?

File: sunbird-rc-esignet-integration-impl/postman-collections/eSignet-with-Sunbird-RC.postman_collection.json
Comment (ase-101) [inline sunbird-rc-esignet-integration-impl/postman-collections/eSignet-with-Sunbird-RC.postman_collection.json:1]: We should add this collection as a folder is eSignet postman collection

File: sunbird-rc-esignet-integration-impl/pom.xml [hunk 5]
```diff
@@ -0,0 +1,348 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<groupId>io.mosip.esignet.sunbirdrc</groupId>
```
Comment (ase-101) [inline sunbird-rc-esignet-integration-impl/pom.xml:5]: We should change the groupId to "io.mosip.esignet.plugin.sunbirdrc"

File: sunbird-rc-esignet-integration-impl/pom.xml [hunk 6]
```diff
@@ -0,0 +1,348 @@
...
+		<url>https://github.com/mosip/digital-credential-plugins</url>
+		<tag>HEAD</tag>
+	</scm>
+	<developers>
+		<developer>
+			<name>Mosip</name>
+			<email>mosip.emailnotifier@gmail.com</email>
+			<organization>io.mosip</organization>
+			<organizationUrl>https://github.com/mosip/digital-credential-plugins</organizationUrl>
+		</developer>
```
Comment (ase-101) [inline sunbird-rc-esignet-integration-impl/pom.xml:32]: It should be changed in accordance with the new repository

File: sunbird-rc-esignet-integration-impl/pom.xml [hunk 7]
```diff
@@ -0,0 +1,348 @@
...
+			<groupId>org.projectlombok</groupId>
+			<artifactId>lombok</artifactId>
+			<scope>compile</scope>
+			<version>1.18.22</version>
+		</dependency>
+
+		<dependency>
+			<groupId>io.mosip.esignet</groupId>
+			<artifactId>esignet-integration-api</artifactId>
+			<version>1.2.0</version>
```
Comment (ase-101) [inline sunbird-rc-esignet-integration-impl/pom.xml:70]: We should change this to latest version of esignet-inegration-api

File: sunbird-rc-esignet-integration-impl/pom.xml [hunk 8]
```diff
@@ -0,0 +1,348 @@
...
+			<scope>provided</scope>
+		</dependency>
+		
+		<dependency>
+			<groupId>io.mosip.kernel</groupId>
+			<artifactId>kernel-keymanager-service</artifactId>
+			<version>1.2.0.1</version>
```
Comment (ase-101) [inline sunbird-rc-esignet-integration-impl/pom.xml:77]: Change this to 1.2.1.0

File: sunbird-rc-esignet-integration-impl/src/main/java/io/mosip/esignet/sunbirdrc/integration/service/SunbirdRCAuthenticationService.java [hunk 9]
```diff
@@ -0,0 +1,199 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+package io.mosip.esignet.sunbirdrc.integration.service;
```
Comment (ase-101) [inline sunbird-rc-esignet-integration-impl/src/main/java/io/mosip/esignet/sunbirdrc/integration/service/SunbirdRCAuthenticationService.java:6]: Package name should be changed to "io.mosip.esignet.plugin.sunbirdrc"

File: sunbird-rc-esignet-integration-impl/README.md
Comment (sacrana0) [reply sunbird-rc-esignet-integration-impl/README.md:1]: Should we also change the artifactId and project name to 'sunbird-rc-plugin' in pom.xml?

File: sunbird-rc-esignet-integration-impl/README.md (hunk 4 above)
Comment (sacrana0) [reply sunbird-rc-esignet-integration-impl/README.md:52]: I am not sure of the compatibility of sunbird-rc with eSignet. Can you please specify what to change it to exactly?

File: sunbird-rc-esignet-integration-impl/postman-collections/eSignet-with-Sunbird-RC.postman_collection.json
Comment (sacrana0) [reply sunbird-rc-esignet-integration-impl/postman-collections/eSignet-with-Sunbird-RC.postman_collection.json:1]: I will move it to esignet, will create a pr. After that is merged, will remove it from here in a new pr. Is that, ok?

File: sunbird-rc-esignet-integration-impl/pom.xml (hunk 5 above)
Comment (sacrana0) [reply sunbird-rc-esignet-integration-impl/pom.xml:5]: the groupId of this plugin is inconsistent with groupid of other plugins. Is that ok?
mock-identity-plugin and mock-plugin have "io.mosip.esignet" as groupid.

**Comment by sacrana0** [comment]
Hi @ase-101 I need some input on some of the changes you want me to make. Please check the comments.

**Review by ase-101 — APPROVED** [review]
